src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,52 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static allAnimals = [];
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    this.hidden = false;
+    Animal.allAnimals.push(this);
+    Animal.alive.push(this);
+  }
+
+  static updateAliveAnimals() {
+    Animal.alive = Animal.allAnimals.filter((animal) => animal.health > 0);
+  }
```

Comment:
The `updateAliveAnimals` static method seems redundant since you are already filtering dead animals in the `checkIfAlive` method. Consider removing this method to simplify your code and avoid duplicating the logic for updating the `Animal.alive` array.